webapp/src/components/modals/channel_settings/select_channel_subscription.tsx

```diff
@@ -39,6 +56,40 @@ export default class SelectChannelSubscriptionInternal extends React.PureCompone
             );
         }
 
+        const subscriptionRows = this.props.channelSubscriptions.map((sub) => (
+            <div
+                key={sub.id}
+                className='select-channel-subscriptions-row'
+            >
+                <ConfirmModal
```

Comment:
I wonder if we can have the confirm modal rendered and managed once in `channel_settings_internal.tsx`, to remove deplicated code. The function passed down to `EditChannelSettings` and `SelectChannelSubscription` would take the subscription to be deleted as a parameter, and stored in the state of the component temporarily while the modal is open. This would relieve the "ConfirmModal per subscription" loop here.

I think this would clean up the code nicely. I'd be happy to talk about this if it's not clear.

Reply:
> The function passed down to EditChannelSettings and SelectChannelSubscription would take the subscription to be deleted as a parameter, and stored in the state of the component temporarily while the modal is open.

This component is already the parent that stores this in its state. So it wouldn't necessarily need to take it as a parameter. This change would involve changing tests too.